runtime/local/src/lib.rs

```diff
@@ -775,17 +775,78 @@ impl pallet_sudo::Config for Runtime {
 )]
 pub enum ProxyType {
     Any,
+    NonTransfer,
+    Governance,
+    CancelProxy,
+    DappsStaking,
 }
 
 impl Default for ProxyType {
     fn default() -> Self {
         Self::Any
     }
 }
+
 impl InstanceFilter<RuntimeCall> for ProxyType {
-    fn filter(&self, _c: &RuntimeCall) -> bool {
+    fn filter(&self, c: &RuntimeCall) -> bool {
         match self {
             ProxyType::Any => true,
+            ProxyType::NonTransfer => {
+                matches!(
+                    c,
+                    RuntimeCall::System(..)
+                        | RuntimeCall::Utility(..)
+                        | RuntimeCall::Timestamp(..)
+                        | RuntimeCall::Grandpa(..)
+                        // Skip entire Balances pallet
+                        | RuntimeCall::Vesting(pallet_vesting::Call::vest{..})
+				        | RuntimeCall::Vesting(pallet_vesting::Call::vest_other{..})
+				        // Specifically omitting Vesting `vested_transfer`, and `force_vested_transfer`
+                        | RuntimeCall::DappsStaking(..)
+                        // Skip entire EVM pallet
+                        // Skip entire Ethereum pallet
+                        // Skip entire EthCall pallet
+                        | RuntimeCall::BaseFee(..)
+                        // Skip entire Contracts pallet
+                        // Skip entire Assets pallet
+                        | RuntimeCall::Scheduler(..)
+                        | RuntimeCall::Democracy(..)
+                        | RuntimeCall::Council(..)
+                        | RuntimeCall::TechnicalCommittee(..)
+                        | RuntimeCall::Treasury(..)
+                        | RuntimeCall::Proxy(..)
+                        | RuntimeCall::Xvm(..)
+                )
```

Comment:
Wouldn't it be better to specify what shouldn't be matched instead?
Seems very brittle like it is right now.

Just a thought.

Reply:
That's definitely one thing we can consider.
What I thought is that whitelisting is more secure than blacklisting, since we know what exactly calls are available for each proxy types without having a chance to introduce unexpected calls which shouldn't be included, especially when devs increase. 
But as long as using allow all calls under a pallet `(..)`, there's a chance.